Gordon360/Services/NewsService.cs

```diff
@@ -153,9 +184,161 @@ public StudentNewsViewModel EditPosting(int newsID, StudentNews newData)
             newsItem.Subject = newData.Subject;
             newsItem.Body = newData.Body;
 
+            if (newData.Image != null)
+            {
+                UploadImage(newData.Image, newsID);
+            }
+
+            //If the image property is null, it means the user removed whatever
+            //picture they had in their posting before.
+            else
+            {
+                RemoveImage(newsItem.SNID);
+                newsItem.Image = newData.Image;
+            }
+
             _unitOfWork.Save();
             
-            return (StudentNewsViewModel)newsItem;
+            return newsItem;
+        }
+
+        /// <summary>
+        /// Takes a filepath for an image, navigates to it, collects the raw data
+        /// of the file and converts it to base64 format. 
+        /// 
+        /// The base64 data will not include the first part of a base64 image ("data:image/...").
+        /// This is because this part is removed in every image before being submitted, and it
+        /// is readded in the frontend before being displayed.
+        /// 
+        /// This helper function does not perform any error checking; every place that calls it
+        /// checks that the path is not empty. Theoretically if it isn't empty it's certainly a valid path.
+        /// </summary>
+        /// <param name="imagePath">The path to the image</param>
+        /// <returns>The base64 content of the image</returns>
+        private string GetBase64ImageDataFromPath(string imagePath)
+        {
+            using (System.Drawing.Image image = System.Drawing.Image.FromFile(imagePath))
+            using (MemoryStream data = new MemoryStream())
+            {
+                image.Save(data, image.RawFormat);
+                byte[] imageBytes = data.ToArray();
+                string base64Data = System.Convert.ToBase64String(imageBytes);
+                return base64Data;
+            }
+        }
+
+        /// <summary>
+        /// Iterates through a collection of news items that have just been queried from
+        /// the database, replacing the Image entry of each (which contains the path to an image)
+        /// with the image itself. This is done because when the frontend makes requests for news items,
+        /// it needs the image data and not the path, but we store the path in the DB for efficiency and
+        /// storage reasons.
+        /// </summary>
+        /// <param name="items">The collection of student news entries</param>
+        /// <returns>The collection of entries with base64 data in their image variable</returns>
+        private IEnumerable<StudentNewsViewModel> InsertImageData(IEnumerable<StudentNewsViewModel> items)
```

Comment:
There's a potentially more idiomatic way of achieving this using an implicit conversion operator. Since you retrieve objects of the `StudentNews.cs` class from the database but return objects of the `StudentNewsViewModel.cs` class to the front end, you could have the conversion from `StudentNews` to `StudentNewsViewModel` handle loading the image data from the path automatically.